include/igl/writeOBJ.cpp

```diff
@@ -118,6 +118,35 @@ IGL_INLINE bool igl::writeOBJ(
   return true;
 }
 
+template <typename DerivedV>
+IGL_INLINE bool igl::writeOBJ(
+  const std::string str,
+  const Eigen::MatrixBase<DerivedV>& V,
+  const std::vector<std::vector<typename DerivedV::Index> >& F)
+{
+  using namespace std;
+  using namespace Eigen;
+  assert(V.cols() == 3 && "V should have 3 columns");
+  ofstream s(str);
+  if(!s.is_open())
+  {
+    fprintf(stderr,"IOError: writeOBJ() could not open %s\n",str.c_str());
+    return false;
+  }
+  s<<V.format(IOFormat(FullPrecision,DontAlignCols," ","\n","v ","","","\n"));
+  
+  for(const auto& face : F) {
+    int face_size = face.size();
+    assert(face_size == 3 || face_size == 4);
```

Comment:
`.obj` can support polygonal faces with > 4 vertices. We could also support line graphs, by replacing the prefix `f` with `l` in case `face_size == 2`.

Reply:
Good point. It turns out face statements with two vertices (edges) are supported by some programs (Blender). So it might make sense to allow that.

Reply:
Actually, meshlab will open a .obj where edges are marked using the `l` marker, but will complain that faces should have `>= 3` vertices if you use `l`. I'm pretty sure blender supports both, so I would rather use `l` if `face_size == 2`.